include/flamegpu/runtime/DeviceAPI.cuh

```diff
@@ -343,6 +431,17 @@ __device__ T ReadOnlyDeviceAPI::getVariable(const char(&variable_name)[M], const
     return value;
 }
 
+template<typename MessageIn, typename MessageOut>
+template<typename T, unsigned int N>
+__device__ void DeviceAPI<MessageIn, MessageOut>::setVariable(const char(&variable_name)[N], T value) {
+    if (variable_name[0] == '_') {
```

Comment:
Should this DTHROW if setabelts is on instead?

Reply:
Yes, but that's a separate issue affecting whole of DeviceAPI. Will create a separate issue/pr.